### DISCUSSION TITLE
Loading bar

### DISCUSSION THREAD

**Discussion by pajhiwo** [czerus/pydriverr#59] [Ideas]
@czerus 
What do you thing about somethign like this?
I think it could be extende do unziping or deletion as well

https://user-images.githubusercontent.com/4226632/111365439-6733f800-868a-11eb-9fd7-1e9c7d3b248a.mp4

**Comment by czerus** [comment]
My initial idea was to add some kind of spinner only to indicate that script is still running and not hanged during download from network like:
![spinner](https://user-images.githubusercontent.com/4174919/111621702-96f91200-87e8-11eb-9f17-3772202cb548.gif)
But now I am more towards overhauling how the logging works. I mean most of important data (also in case of failure) goes only to log file. Log file is OK but now I would change it in following way:
* clearly separated step - most probably more than now 
* marking each step as completed

Something like: 
![spinner_new](https://user-images.githubusercontent.com/4174919/111622614-b3497e80-87e9-11eb-80bf-e837f088f8e8.gif)
This could also contains download info [how many out of total] about downloaded but smaller that presented by you.
So I believe it would be bigger task that would structurize logs and make it more pretty?
If you still want to do it. That is OK. new fresh eye and ideas are OK for me.

**Comment by pajhiwo** [comment]
that lib you showed does not worrk ok win windows and you cannot change spinner from lines to dots.
i have played with https://github.com/pavdmyt/yaspin witch is very simillar
take a look on example and let me know what you think


https://user-images.githubusercontent.com/4226632/113426382-4af2c380-93cb-11eb-9f7b-0fafa5f2e2af.mp4

**Reply by czerus** [reply]
yeh. it s much better now. Only one thing is missing for me in those prints:
in case of update, install, delete I would like to see full info bout given webdriver: version, or and arch.

**Comment by czerus** [comment]
@pajhiwo please take a look here: https://click.palletsprojects.com/en/8.0.x/api/?highlight=echo#click.progressbar
Looks like click (which we use internally) supports progressbar

**Reply by pajhiwo** [reply]
hmm...from what i see it required iterable element or something with "chunks" (parts) to update progress bar manually. so it cannot be used directly in pydriver.py. instead we would have to push click lib inside other files.

**Comment by pajhiwo** [comment]
I have checkec yaspisn lib (the one showd before) and it seems to be working.
We can do it in different ways:
1. Have one spinner ex. "Installing" that will be animatid and on end will show all logger.info messages and "done" state for installing
![2021-06-20 13_02_13-cmd - poetry  shell](https://user-images.githubusercontent.com/4226632/122673397-2affbe80-d1c8-11eb-8ccd-9282c1167448.png)

2. We can have spinners for different stages ex. downlading, installing, etc:
![2021-06-20 13_02_32-cmd - poetry  shell](https://user-images.githubusercontent.com/4226632/122673422-45399c80-d1c8-11eb-96bd-3bb1ab364825.png)

3. We can have only spinner messages whithout showind info from logger.info to console to keep console (clean)
![2021-06-20 13_02_22-cmd - poetry  shell](https://user-images.githubusercontent.com/4226632/122673441-639f9800-d1c8-11eb-9d38-1b926675a1fe.png)

Please comment.

**Reply by czerus** [reply]
I like options 2 and 3. 3rd is cleanest one. So maybe lets go with it. What do you think?
If we will choose 3rd one how does logging to console will look like. Will we keep detailed log in file? How messages on console will be like when errors appears?
